### PR TITLE
Support corporate proxy servers

### PR DESCRIPTION
**Pull request by jlanng**
See issue #6562

I've tested this fix behind a corporate proxy, with an explicit proxy parameter specified, and with a direct connection.

### CODE REVIEW

**Comment by livarcocc** [comment]
@dotnet-bot Test Windows_NT x86 Debug Build

File: scripts/obtain/dotnet-install.ps1 [hunk 1]
```diff
@@ -140,6 +140,17 @@ function GetHTTPResponse([Uri] $Uri)
     try {
         # HttpClient is used vs Invoke-WebRequest in order to support Nano Server which doesn't support the Invoke-WebRequest cmdlet.
         Load-Assembly -Assembly System.Net.Http
+ 
+        if(-not $ProxyAddress)
```
Comment (livarcocc) [inline scripts/obtain/dotnet-install.ps1]: I imagine we need something similar in the sh version of the script.

**Comment by livarcocc** [comment]
cc @johnbeisner

File: scripts/obtain/dotnet-install.ps1 (hunk 1 above)
Comment (omajid) [inline scripts/obtain/dotnet-install.ps1]: I think my comment on https://github.com/dotnet/cli/pull/5865 applies here too: nothing in `dotnet-install.sh` refers to proxy servers and users can use `http_proxy/HTTP_PROXY` environment variables to set the proxy directly. Using `~/.curlrc` is also an option.

Comment (jlanng) [inline scripts/obtain/dotnet-install.ps1]: @omajid  yes I agree, that's the standard way to do it on Linux platforms so no need to do anything on the .sh side

**Comment by livarcocc** [comment]
@MattGertz for approval.

**Comment by MattGertz** [comment]
Approved & thank you!

**Pull request merged**